### PR TITLE
Run-time data is deleted after the workflow is completed

### PR DESCRIPTION
**Pull request by grfeng**
Run-time data is deleted after the workflow is completed and history is stored in the ES
#415

### CODE REVIEW

File: core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java [hunk 1]
```diff
@@ -345,6 +345,8 @@ public void completeWorkflow(Workflow wf) throws Exception {
         }
         Monitors.recordWorkflowCompletion(workflow.getWorkflowType(), workflow.getEndTime() - workflow.getStartTime(), wf.getOwnerApp());
         queueDAO.remove(deciderQueue, workflow.getWorkflowId());    //remove from the sweep queue
+
```
Comment (pctreddy) [inline core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java]: Can this be a configuration and runtime data deleted only if the configuration is set to that value ?
Conductor Config looks like a good place for it and it does not break the functionality where the users might be still using the runtime data from some other systems to query.

File: core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java [hunk 2]
```diff
@@ -505,15 +507,15 @@ public void updateTask(TaskResult result) throws Exception {
...
-        decide(workflowId);
-
         if (task.getStatus().isTerminal()) {
             long duration = getTaskDuration(0, task);
             long lastDuration = task.getEndTime() - task.getStartTime();
             Monitors.recordTaskExecutionTime(task.getTaskDefName(), duration, true, task.getStatus());
             Monitors.recordTaskExecutionTime(task.getTaskDefName(), lastDuration, false, task.getStatus());
         }
 
+        decide(workflowId);
```
Comment (pctreddy) [inline core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java]: Is this moved here to fix the  recording of the metrics ?

File: redis-persistence/src/main/java/com/netflix/conductor/dao/dynomite/RedisExecutionDAO.java [hunk 3]
```diff
@@ -378,7 +378,7 @@ public Workflow getWorkflow(String workflowId, boolean includeTasks) {
 			throw new ApplicationException(Code.NOT_FOUND, "No such workflow found by id: " + workflowId);
 		}
 		Workflow workflow = readValue(json, Workflow.class);
```
Comment (pctreddy) [inline redis-persistence/src/main/java/com/netflix/conductor/dao/dynomite/RedisExecutionDAO.java]: Same as above can be driven by a config parameter from the Conductor config.

File: test-harness/src/test/java/com/netflix/conductor/tests/utils/MockIndexDAO.java [hunk 4]
```diff
@@ -18,8 +18,8 @@
  */
 package com.netflix.conductor.tests.utils;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.util.*;
```
Comment (pctreddy) [inline test-harness/src/test/java/com/netflix/conductor/tests/utils/MockIndexDAO.java]: One of the things that will be added to the Conductor code base in near future is `FindBugs` and `PMD`, which will help catch the smaller static analysis sooner. Until then please take out the import with *

File: core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java (hunk 1 above)
Comment (grfeng) [reply core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java]: Ok, However it is available from Elasticsearch.

File: core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java (hunk 2 above)
Comment (grfeng) [reply core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java]: Yes, `decide(workflowId)` will be remove runtime data.

**Comment by grfeng** [comment]
@pctreddy Any problem？

**Pull request closed without merging**